pandas/core/frame.py

```diff
@@ -8288,6 +8286,8 @@ def _construct_result(self, result) -> DataFrame:
         #  non-unique columns case
         out.columns = self.columns
         out.index = self.index
+        if not getattr(self, "attrs", None) and getattr(other, "attrs", None):
+            out = out.__finalize__(other)
```

Comment:
```suggestion
        out = out.__finalize__(other)
```

Appears `__finalize__` will correctly skip if `other` has a populated `attrs`

Reply:
I think this line `__finalize__` needs a fix:

`self.flags.allows_duplicate_labels = other.flags.allows_duplicate_labels`

Prioritizing `False` if  `self.flags.allows_duplicate_labels` or `other.flags.allows_duplicate_labels` is `False`

Reply:
What about doing in `__finalize__`
```python
if isinstance(other, NDFrame):
    if other.attrs:
        # We want attrs propagation to have minimal performance
        # impact if attrs are not used; i.e. attrs is an empty dict.
        # One could make the deepcopy unconditionally, but a deepcopy
        # of an empty dict is 50x more expensive than the empty check.
        self.attrs = deepcopy(other.attrs)
    self.flags.allows_duplicate_labels = (
        self.flags.allows_duplicate_labels
        and other.flags.allows_duplicate_labels
    )
```

Reply:
Yup that's the correct location to fix this